modules/debounce.js

```diff
@@ -1,35 +1,41 @@
 import restArguments from './restArguments.js';
-import delay from './delay.js';
+import now from './now.js';
 
 // When a sequence of calls of the returned function ends, the argument
 // function is triggered. The end of a sequence is defined by the `wait`
 // parameter. If `immediate` is passed, the argument function will be
 // triggered at the beginning of the sequence instead of at the end.
 export default function debounce(func, wait, immediate) {
-  var timeout, result;
+  var timeout, previous, args, result, context;
 
-  var later = function(context, args) {
-    timeout = null;
-    if (args) result = func.apply(context, args);
-  };
+  var later = function() {
+    var passed = now() - previous;
+    if (wait >= passed) {
```

Comment:
Shouldn't this be `>`? If `wait == passed`, I'd think you want to invoke immediately instead of setting another timeout.

Reply:
You're absolutely right.

Reply:
I have made a version in parallel with this that does not use timeout when immediate == true, so it confused me, since I there had the opposite order in the if else block. Good thing you catched it!